app/utils/safe-await.ts

```diff
@@ -0,0 +1,35 @@
+// TypeScript port of github.com/DavidWells/safe-await/
+
+// Native Error types https://mzl.la/2Veh3TR
+const nativeExceptions = [
+  EvalError,
+  RangeError,
+  ReferenceError,
+  SyntaxError,
+  TypeError,
+  URIError,
+].filter(except => typeof except === 'function');
+
+function throwNative(error: Error) {
+  for (const Exception of nativeExceptions) {
+    if (error instanceof Exception) throw error;
+  }
+}
+
+export function safeAwait<T>(promise: Promise<T>, finallyFn?: () => any) {
```

Comment:
Nice :+1:

Reply:
Maybe we can put in ui lib even as a helper? Really does tidy up the code a lot.